### ISSUE TITLE
Compare releases agains previous release (skipping prerelease)

### ISSUE THREAD

**Issue by burtek** [absolute-version/commit-and-tag-version#138]
Right now when I create a release after a pre-release, the release changelo only contains changes since the pre-release and the compare link in changelog points to diff between pre-release and this release.

I'd like to make releases always compare against last full release rather than prerelease.

Let's say I have versions `1.0.0`, `1.1.0-alpha.1`, `1.1.0-alpha.2` and `1.1.0`.

I want the changelog to show:
- for `1.1.0-alpha.1` - changes since `1.0.0` (as it works currently)
- for `1.1.0-alpha.2` - changes since `1.1.0-alpha.1` (as it works currently)
- for `1.1.0` - changes since `1.0.0`

Is this possible now?

**Comment by TimothyJones** [comment]
I don't think this is currently possible without mucking about with multiple sets of tags (or renaming tags out of the way, which I don't think is a good idea).

If I had this use case once or twice, I would just delete the pre-release tags locally and then run `commit-and-tag-version`. If it was often, something more automated might be better

**Comment by burtek** [comment]
So for now I'm gonna make a script that removes prerelease tags, runs this library and pulls tags back from origin.

But I guess that's something that can possibly be implemented and end up being a PR here?

**Comment by TimothyJones** [comment]
The [rationale for which changes to accept is here](https://github.com/absolute-version/commit-and-tag-version/discussions/30) - I feel like it fits, although I'm not sure that it would be easy to come up with a general case that doesn't make assumptions about how / when the tool is invoked. What do you think?

**Comment by dirkluijk** [comment]
Also interested in a better solution, I now have to manually remove tags to recreate all the proper changes in the change log after pre-releases.

What about filtering out all the `x.y.z.*` tags when it does not run in pre-release mode?

**Comment by dirkluijk** [comment]
@burtek can you maybe share the script you used?

**Comment by burtek** [comment]
Nothing sofisticated really @dirkluijk 

```sh
for tag in $(git tag | grep -E '^v[0-9]+\.[0-9]+\.[0-9]+-(alpha|beta)\.[0-9]+$'); do
    git tag -d "$tag"
done

yarn release # runs commit-and-tag-version

git fetch --tags
```
